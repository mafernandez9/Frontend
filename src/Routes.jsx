import React from 'react';
import { Route, Switch } from 'react-router-dom';
// import Home from './views/Home';
import UserList from './views/UserList';
import UserDetail from './views/UserDetail';
import UserProfile from './views/Profile';
import UserEdit from './views/UserEdit';
import UserDelete from './views/UserDelete';
import NotFound from './views/NotFound';
import BookList from './views/BookList';
import BookDetail from './views/BookDetail';
import BookEdit from './views/BookEdit';
import BookDelete from './views/BookDelete';
import Login from './views/Login';
import NavBar from './components/NavBar';
import Footer from './components/Footer';
// import LikesNew from './components/LikesNew';
import ReviewDetail from './views/ReviewDetail';
import ReviewDelete from './views/ReviewDelete';
// import ReviewEdit from './views/ReviewEdit';
import BookNew from './components/BookNew';
import ReviewNew from './components/ReviewNew';
import NewLike from './components/NewLike';
import NewReport from './components/NewReport';
import ReviewEdit2 from './views/ReviewEdit';
import UserReviews from './views/UserReviews';
import Register from './views/Register';
import AuthContextProvider from './contexts/AuthContext';

export default function Routes() {
  return (
    <AuthContextProvider>
      <NavBar />
      <Switch>
        <Route exact path="/" component={BookList} />
        <Route exact path="/users" component={UserList} />
        <Route exact path="/users/me" component={UserProfile} />
        <Route exact path="/users/:id" component={UserDetail} />
        <Route exact path="/users/:id/reviews" component={UserReviews} />
        <Route exact path="/users/:id/edit" component={UserEdit} />
        <Route exact path="/users/:id/delete" component={UserDelete} />
        <Route exact path="/books" component={BookList} />
        <Route exact path="/books/:id" component={BookDetail} />
        <Route exact path="/books/:id/edit" component={BookEdit} />
        <Route exact path="/books/:id/delete" component={BookDelete} />
        <Route exact path="/newbook" component={BookNew} />
        <Route exact path="/books/:id/newreview" component={ReviewNew} />
        <Route exact path="/login" component={Login} />
        <Route exact path="/signup" component={Register} />
        <Route exact path="/books/:bookId/reviews/:reviewId" component={ReviewDetail} />
        <Route exact path="/books/:bookId/reviews/:reviewId/delete" component={ReviewDelete} />
        {/* <Route exact path="/books/:bookId/reviews/:reviewId/likes" component={LikesNew} /> */}
        <Route exact path="/books/:bookId/reviews/:reviewId/like" component={NewLike} />
        <Route exact path="/books/:bookId/reviews/:reviewId/report" component={NewReport} />
        <Route exact path="/books/:bookId/reviews/:reviewId/edit" component={ReviewEdit2} />
        <Route component={NotFound} />
      </Switch>
      <Footer />
    </AuthContextProvider>
  );
}
