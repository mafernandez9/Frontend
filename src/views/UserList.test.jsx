import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import axios from 'axios';
import Routes from '../Routes';

jest.mock('axios');

function TestRouter({ path }) {
  return (
    <MemoryRouter initialEntries={[path]}>
      <Routes />
    </MemoryRouter>
  );
}

const userResponse = {
  data:
    [
      {
        type: 'users',
        id: '2',
        attributes:
            {
              firstName: 'Jakayla',
              lastName: 'Smitham',
              birthDay: '1994-05-21',
              nickname: 'Elody',
              email: 'Liza30@hotmail.com',
              image: 'https://picsum.photos/200',
              role: 'client',
            },
      },
    ],
};

describe('UserList', () => {
  describe('when user gets into the page', () => {
    it('renders the home page with links to signup and log in', async () => {
      axios.get.mockImplementationOnce(() => Promise.resolve(userResponse));
      const tree = renderer.create(<TestRouter path="/users" />);
      await Promise.resolve();
      expect(tree.toJSON()).toMatchSnapshot();
    });
  });
});