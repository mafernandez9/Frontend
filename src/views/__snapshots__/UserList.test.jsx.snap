// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserList when user gets into the page renders the home page with links to signup and log in 1`] = `
Array [
  <nav
    className="nav-bar"
  >
    <span
      className="logo"
    >
      <h3
        className="bookkers"
      >
        <a
          className="link"
          href="/"
          onClick={[Function]}
        >
          ðŸ“šBookkers
        </a>
      </h3>
    </span>
    <span
      className="options"
    >
      <a
        className="link"
        href="/login"
        onClick={[Function]}
      >
        <p>
          Log in
        </p>
      </a>
      <a
        className="link"
        href="/signup"
        onClick={[Function]}
      >
        <p>
          Sign up
        </p>
      </a>
    </span>
  </nav>,
  <div>
    <div
      className="users"
    >
      <div
        id="title-comunidad"
      >
        <h3
          className="title"
        >
          Comunidad
        </h3>
      </div>
      <div
        className="search-bar"
      >
        <input
          placeholder="Â¿QuÃ© buscas?"
        />
        <button
          type="submit"
        >
          Buscar
        </button>
      </div>
      <div
        className="user-list"
      />
    </div>
  </div>,
  <footer>
    ðŸ“š Bookkers hecha por: Paula Torres, MatÃ­as FernÃ¡ndez y Marjorie BascuÃ±Ã¡n.
  </footer>,
]
`;
